clash-lib/src/Clash/Core/TermLiteral.hs

```diff
@@ -47,6 +51,24 @@ class Typeable a => TermLiteral a where
     -- ^ 'Left' indicates a failure, containing the (sub)term that failed to
     -- translate. 'Right' indicates a success.
 
+  -- | Pretty print the type of a term (for error messages). Its default implementation
+  -- uses 'Typeable' to print the type. Note that this method is there to allow
+  -- an instance for 'SNat' to exist (and other GADTs imposing 'KnownNat').
+  -- Without it, GHC would ask for a 'KnownNat' constraint on the instance, which
+  -- would defeat the purpose of it.
+  termTypePrec ::
+    -- | The operator precedence of the enclosing context (a number from @0@ to
+    -- @11@). Function application has precedence @10@. Used to determine whether
+    -- the result should be wrapped in parentheses.
+    Int ->
+    -- | Term whose type needs to be pretty printed
+    Proxy a ->
+    -- | Pretty printed type
```

Comment:
It's [more complicated](https://hackage.haskell.org/package/base-4.17.0.0/docs/Prelude.html#t:ShowS) then that.
Maybe just leave out the description of the result?
I'd say it's not worth it to describe it here.